lib/invoices/utils.ts

```diff
@@ -242,3 +242,296 @@ const pdfDefinition = (input: {
+  }
+  const address = application.shippingAddressSameAsHomeAddress
+    ? {
+        addressLine1: application.addressLine1,
+        addressLine2: application.addressLine2,
+        city: application.city,
+        province: application.province,
+        country: application.country,
+        postalCode: application.postalCode,
+      }
+    : {
+        addressLine1: application.shippingAddressLine1 as string,
+        addressLine2: application.shippingAddressLine2,
+        city: application.shippingCity as string,
+        province: application.shippingProvince as string,
+        country: application.shippingCountry as string,
+        postalCode: application.shippingPostalCode as string,
+      };
+
+  const definition = donationPdfDefinition({
+    applicantName,
+    userNumber: applicantId,
+    appNumber: appNumber,
+    permitType,
+    receiptNumber,
+    dateIssued: new Date(),
+    dateDonationRecevied: createdAt,
+    issuedBy: employeeInitials,
+    donationAmount,
+    paymentItems,
+    totalAmount,
+    address,
+    nonNullEmail,
+  });
+  const printer = new pdfPrinter({
+    Helvetica: {
+      normal: 'Helvetica',
+      bold: 'Helvetica-Bold',
+      italics: 'Helvetica-Oblique',
+      bolditalics: 'Helvetica-BoldOblique',
+    },
+  });
+
+  const pdfDocument = printer.createPdfKitDocument(definition);
+  pdfDocument.end();
+  return pdfDocument;
+};
+
+/** PDF generation schema */
+const donationPdfDefinition = (input: {
+  applicantName: string;
+  userNumber: number | null;
+  appNumber: number | null;
+  permitType: string;
+  receiptNumber: string;
+  dateIssued: Date;
+  dateDonationRecevied: Date;
+  issuedBy: string;
+  paymentItems: Array<{
+    item: string;
+    amount: Prisma.Decimal;
+    paidBy: PaymentType;
+    subtotal: Prisma.Decimal;
+  }>;
+  totalAmount: Prisma.Decimal;
+  donationAmount: Prisma.Decimal;
+  address: {
+    addressLine1: string;
+    addressLine2: string | null;
+    city: string;
+    province: string;
+    country: string;
+    postalCode: string;
+  };
+  nonNullEmail: string;
+}): any => {
+  const {
+    applicantName,
+    userNumber,
+    appNumber,
+    permitType,
+    receiptNumber,
+    donationAmount,
+    totalAmount,
+    dateIssued,
+    issuedBy,
+    dateDonationRecevied,
+    paymentItems,
+    address,
+    nonNullEmail,
+  } = input;
+
+  return {
+    footer: function (currentPage: number, pageCount: number) {
+      return currentPage == pageCount
+        ? {
+            text: `For information on all registered charities in Canada under the Income Tax Act please contact: Canada Revenue Agency www.cra.gc.ca/charities-giving `,
+            style: 'footer',
+          }
+        : null;
+    },
+    content: applicationPdfDefinition({
+      applicantName,
+      userNumber,
+      permitType,
+      receiptNumber,
+      dateIssued,
+      issuedBy,
+      paymentItems,
+      totalAmount,
+      address,
+    }).content.concat([
```

Comment:
This is the invoice PDF. If we have a donation, we include both the invoice and donation PDFs.